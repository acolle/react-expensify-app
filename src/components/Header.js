import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => (
  <header>
    <h1>Expensify App</h1>
    <ul>
      <li><NavLink to="/" activeClassName="is-active" exact={true}>Home</NavLink></li>
      <li><NavLink to="/create" activeClassName="is-active">Add expense</NavLink></li>
      <li><NavLink to="/edit" activeClassName="is-active">Edit expense</NavLink></li>
      <li><NavLink to="/help" activeClassName="is-active">Help</NavLink></li>
    </ul>
  </header>
);

export default Header;
