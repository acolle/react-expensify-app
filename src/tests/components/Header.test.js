import React from 'react';
import { shallow } from 'enzyme';
// import toJSON from 'enzyme-to-json'; // no need as it's defined in jest.config.json
// import ReactShallowRenderer from 'react-test-renderer/shallow';
import Header from '../../components/Header';

test('should render Header correctly', () => {
  const wrapper = shallow(<Header />);
  expect(wrapper).toMatchSnapshot();
  // expect(wrapper.find('h1').text()).toBe('Expensify App');
  // const renderer = new ReactShallowRenderer();
  // renderer.render(<Header />); // output we're trying to test
  // expect(renderer.getRenderOutput()).toMatchSnapshot();
});
