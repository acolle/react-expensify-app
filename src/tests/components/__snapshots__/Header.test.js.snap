// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header>
  <h1>
    Expensify App
  </h1>
  <ul>
    <li>
      <NavLink
        activeClassName="is-active"
        exact={true}
        to="/"
      >
        Home
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="is-active"
        to="/create"
      >
        Add expense
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="is-active"
        to="/edit"
      >
        Edit expense
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="is-active"
        to="/help"
      >
        Help
      </NavLink>
    </li>
  </ul>
</header>
`;
